### PR TITLE
number of investors fix

### PR DESCRIPTION
**Pull request by AlexeyTuboltsev**
this pr fixes NaN showing up instead of number of investors in all relevant places

### CODE REVIEW

File: app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts [hunk 1]
```diff
@@ -69,4 +70,14 @@ describe("Incoming payout", function(): void {
       cy.get(tid("incoming-payout-ether-token")).should("not.exist");
     });
   });
+  it("Should show the number of investors", () => {
```
Comment (desfero) [inline app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts]: add an empty line above for consistency

File: app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts [hunk 2]
```diff
@@ -69,4 +70,14 @@ describe("Incoming payout", function(): void {
...
+    const ETO_ID = etoFixtureAddressByName("ETOInClaimState");
+
+    createAndLoginNewUser({ type: "investor" }).then(() => {
+      goToDashboard();
+      cy.get(tid(`eto-overview-${ETO_ID}-investors-count`)).should($e => {
```
Comment (desfero) [inline app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts]: Let's maybe do more declarative approach 
```
 cy.get(tid(`eto-overview-${ETO_ID}-investors-count`))
        .get(tid("value"))
        .contains("6");
```
What do you think?

File: app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts [hunk 3]
```diff
@@ -69,4 +70,14 @@ describe("Incoming payout", function(): void {
...
+        expect($e.find(tid("value")).text()).to.eq("6");
```
Comment (Moejoe90) [inline app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts]: From where is the value '6' 

We should stop using hardcoded values these fixtures are bound to change without notice so please use the ETO helper function to get the values from the API.

if we have to inquire the ethereum node then we need to write a helper function but that can be in a separate PR

File: app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import {
+  TCompanyEtoData,
+  TEtoSpecsData,
+} from "../../../../../lib/api/eto/EtoApi.interfaces.unsafe";
+import { EEtoSubState, IEtoContractData } from "../../../../../modules/eto/types";
+import { InvestmentLayout } from "./InvestmentStatus";
+
+describe("InvestmentStatus", () => {
+  let initialEnv: any;
+  const testData = cloneDeep(testEto) as TEtoSpecsData & {
```
Comment (desfero) [inline app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx]: Maybe we can add type to reverse `DeepReadonly`? 
```
export type DeepWritable<T> = T extends Primitive | WhitelistedWritableTypes | Function
  ? T
  : T extends (any[] | ReadonlyArray<any>)
  ? WritableArray<T[number]>
  : DeepWritableObject<T>;
type DeepWritableObject<T> = { -readonly [P in keyof T]: DeepWritable<T[P]> };
interface WritableArray<T> extends Array<DeepWritable<T>> {}
```

File: app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    expect(
+      component
+        .find(tid(`eto-overview-${testEto.etoId}-investors-count`))
+        .find(tid("value"))
+        .text(),
+    ).to.eq("100 000 000 000 000 000");
+  });
+
+  it("should throw if number of investors is not parsable as number", () => {
+    testData.contract!.totalInvestment.totalInvestors = "bla";
```
Comment (desfero) [inline app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx]: can we move `testData` to `beforeEach`, otherwise if someone wants to add new tests after this one component will throw because of not parsable

File: app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.tsx [hunk 6]
```diff
@@ -26,7 +26,7 @@ export interface IInvestmentWidgetStateProps {
 
 export type TInvestWidgetProps = IInvestmentWidgetProps & IInvestmentWidgetStateProps;
 
-const InvestmentLayout: React.FunctionComponent<TInvestWidgetProps> = ({ eto }) => {
+export const InvestmentLayout: React.FunctionComponent<TInvestWidgetProps> = ({ eto }) => {
```
Comment (desfero) [inline app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.tsx]: add export to the bottom, like `InvestmentStatus` to keep only single export

File: app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts (hunk 2 above)
Comment (AlexeyTuboltsev) [reply app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts]: done

File: app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts (hunk 1 above)
Comment (AlexeyTuboltsev) [reply app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts]: done

File: app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts (hunk 3 above)
Comment (AlexeyTuboltsev) [reply app/test-e2e/investor-dashboard/LightWalletInvestorDashboard.spec.e2e.io.ts]: done

File: app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx (hunk 4 above)
Comment (AlexeyTuboltsev) [reply app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx]: done

File: app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx (hunk 5 above)
Comment (AlexeyTuboltsev) [reply app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx]: done

File: app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.tsx (hunk 6 above)
Comment (AlexeyTuboltsev) [reply app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.tsx]: ✔️

File: app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+    expect(
+      component
+        .find(tid(`eto-overview-${testEto.etoId}-investors-count`))
+        .find(tid("value"))
+        .text(),
+    ).to.eq("2");
+  });
+
+  it("should show number of investors", () => {
```
Comment (Moejoe90) [inline app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx]: This is not the correct name of the test 
`should show number of investors`
as its already here 
https://github.com/Neufund/platform-frontend/pull/3090/files#diff-cdfe34de963d67d834d48ffa76191affR29

Comment (AlexeyTuboltsev) [reply app/components/eto/overview/EtoOverviewThumbnail/InvestmentStatus/InvestmentStatus.spec.tsx]: fixed

**Comment by glupiawymowka** [comment]
tested on stage
